### ISSUE TITLE
missing documentation for an associated function Builder

### ISSUE THREAD

**Issue by SupperZum** [colin-kiegel/rust-derive-builder#283]
I don't know how to remove this Warning. Please tell me how to remove it.

warning: missing documentation for an associated function
  --> crate\lt\src\lib.rs:30:32
   |
30 | #[ derive( Debug, Deserialize, Builder, Clone, Default ) ]
   |                                ^^^^^^^
   |
note: the lint level is defined here
  --> crate\lt\src\lib.rs:3:11
   |
3  | #![ warn( missing_docs ) ]
   |           ^^^^^^^^^^^^
   = note: this warning originates in the derive macro `Builder` (in Nightly builds, run with -Z macro-backtrace for more info)

**Comment by SupperZum** [comment]
...

**Comment by TedDriggs** [comment]
If you run `cargo doc` on your crate and look at the builder, do you see documentation on the struct or its methods?

You may need to add something like...

```rust
#[derive(Builder)]
#[builder_struct_attr(allow(missing_docs))]
struct Yours {}
```

**Comment by sowbug** [comment]
That exact solution didn't work in my case.

I ended up with code like this in my `cargo expand` output:

```
        #[allow(clippy::all)]
        #[allow(dead_code)]
        impl OrchestratorBuilder {
            /// My rustdoc comment for the title() method
            #[allow(unused_mut)]
            pub fn title(&mut self, value: Option<String>) -> &mut Self {
                ...
            }
            #[allow(unused_mut)]
            pub fn field_that_I_think_was_causing_the_problem(
                &mut self,
                value: ...,
            ) -> &mut Self {
                ...
```

This might be because I'm using `#[builder(setter(skip))]` on the struct, and `#[builder(setter)]` on the field that's acting up. Otherwise it seems like this problem would affect many users of this crate.

A few observations:

1. Putting `#[builder_struct_attr(allow(missing_docs))]` on the struct doesn't cause the Builder's `impl`s to carry the attr, but it does produce it for the Builder struct.
2. I guessed that there was a `#[builder_field_attr(allow(missing_docs))]`, and I tried putting it on `field_that_I_think_was_causing_the_problem`, but the generated Builder code didn't get the attr.
3. The Builder generator appears to copy any rustdoc for a field to its corresponding Builder setter. That's why `title` doesn't generate the missing-doc warning.

Based on Observation #3, I put a placeholder rustdoc on `field_that_I_think_was_causing_the_problem` in the original struct. The warning immediately went away, and indeed the generated Builder impl included a copy of my placeholder rustdoc.

TL;DR: if you're getting this warning, put `/// any rustdoc` on your struct fields, and it'll get propagated to the Builder setters, which will fix the issue.

**Comment by SupperZum** [comment]
My problem was that the structure fields were not documented, when I documented the structure fields, the warning disappeared.
